bip-0077.md

```diff
@@ -0,0 +1,703 @@
+|   (64 bytes)   |                            (7104 bytes)                              |
+|                +-----------------------+---------------------------------+------------+
+|                |       Reply Key       |         Padded Plaintext        |  AEAD Tag  |
+|                |       (33 bytes)      |   (7055 bytes = 7168-64-33-16)  | (16 bytes) |
++---------------------------------------------------------------------------------------+
+```
+
+The resulting HPKE payload is the body of a POST request to the
+receiver's mailbox. This request is then [
+encapsulated](#client-directory-interactions) according to
+Oblivious HTTP to the directory's OHTTP Gateway. OHTTP serializes the
+inner request as BHTTP, and provides another layer of HPKE encryption,
+between the client and directory.
+
+Upon receipt, the directory's OHTTP Gateway decapsulates the OHTTP
+request and handles the inner POST request at the receiver's mailbox
+endpoint, which stores the HPKE encrypted payload to be forwarded to the
+receiver.
+
+The sender then polls OHTTP encapsulated GET requests to the sender's
+mailbox endpoint until it receives a response from the directory
+containing the receiver's *Proposal PSBT*, and proceeds to
+[sign and broadcast](#sender-signing-and-broadcast).
+It stops polling after expiration.
+
+#### Optional sender parameters
+
+[BIP 78's optional sender parameters](https://github.com/bitcoin/bips/blob/master/bip-0078.mediawiki#optional-parameters)
+are also valid as version 2 parameters. Though in version 2 they are included in
+the body as part of the ciphertext rather than as a query string.
+
+HPKE binds ciphertexts to application-specific `info` strings. Because
+of this domain separation, BIP 78's `v` parameter is redundant and
+should be omitted for version 2.
+
+### Receiver Proposal PSBT Messaging
+
+After sharing the Payjoin URI with the sender, the receiver polls via
+OHTTP encapsulated GET requests to the receiver's mailbox endpoint. So
+long as the mailbox contains no message, the directory responds with
+status 202 ACCEPTED. Once a mailbox contains a message, the directory
+returns it in the response body with status 200 OK.
+
+Upon receiving an encapsulated 200 OK response, the receiver decrypts
+the payload and checks the *Original PSBT* therein according to the
+[receiver checklist](#receivers-original-psbt-checklist).
+
+The receiver then updates the *Original PSBT* to include new signed
+inputs and outputs, invalidating the sender's signature(s). The receiver
+may also adjust the transaction fee. The result, called the *Proposal
+PSBT*, must satisfy the [sender checklist](#senders-proposal-psbt-checklist)
+
+The Proposal PSBT MUST:
+
+- Include complete UTXO data.
+- Include all inputs from the Original PSBT.
+- Include all outputs which do not belong to the receiver from the
+  Original PSBT.
+- Insert additional inputs and outputs at a random index.
```

Comment:
Should this be more specific? Is the receiver required to add at least one input? Would it e.g., be permissible to just split his output to perform a cut-through transaction?

Reply:
Okay, in that case, it might make sense to explicitly state that the receiver has to add an input, which seemed implied to me, but I don’t think I saw it mentioned anywhere.

Reply:
The receiver is not required to add an input. They may merely substitute their output, for example, to forward the whole amount to cold storage if it exceeded a certain amount. However, there is no incentive for the sender to prefer such a Proposal PSBT to the Original PSBT beyond cooperation.

Reply:
Either way it may be worth explicitly stating whether or not a receiver has to add an input, at least it was unclear to me after reading the document.

Reply:
addressed by f2ad25b14f8d6da11897a9b9a6f745bf92ead881